### PR TITLE
HS-1490 - Using schemas instead of dbs for services works. Have ignor…

### PR DESCRIPTION
**Pull request by jaymberry**
…ed grafana

## Description
Put db services into different schemas in single db.

## Jira link(s)
- https://issues.opennms.org/browse/HS-1490 

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: charts/opennms/templates/postgres/postgres-init-secret.yaml [hunk 1]
```diff
@@ -7,39 +7,42 @@ metadata:
...
     GRANT ALL ON DATABASE horizon_stream TO opennms;
-  postgres.keycloak.initial.script.sql: |
+
+    \connect horizon_stream;
+
     CREATE USER keycloak with password '{{ .Values.Postgres.KeycloakPassword }}';
-    CREATE DATABASE keycloak;
-    GRANT ALL ON DATABASE keycloak TO keycloak;
-  postgres.inventory.initial.script.sql: |
+    GRANT ALL ON DATABASE horizon_stream TO keycloak;
```
Comment (geraldhumphries) [inline charts/opennms/templates/postgres/postgres-init-secret.yaml:22]: No need to `GRANT ALL` here, `CREATE SCHEMA IF NOT EXISTS keycloak AUTHORIZATION keycloak;` will give the role all the permissions it needs by making it the owner of the schema. It'll also protect against services accessing the wrong schemas.

File: charts/opennms/templates/opennms/inventory/inventory-deployment.yaml [hunk 2]
```diff
@@ -47,7 +47,8 @@ spec:
             - name: JAVA_TOOL_OPTIONS
               value: "-agentlib:jdwp=transport=dt_socket,server=y,address=5005,suspend=n,quiet=y"  # FIXME: Permanent debug port, enable only for dev mode
             - name: SPRING_DATASOURCE_URL
-              value: "jdbc:postgresql://postgres:5432/inventory"
+              #value: "jdbc:postgresql://postgres:5432/inventory"
```
Comment (geraldhumphries) [inline charts/opennms/templates/opennms/inventory/inventory-deployment.yaml:50]: nit: leftover comment

**Review by geraldhumphries — CHANGES_REQUESTED** [review]

File: charts/opennms/templates/postgres/postgres-init-secret.yaml (hunk 1 above)
Comment (jaymberry) [reply charts/opennms/templates/postgres/postgres-init-secret.yaml]: Removed these

File: charts/opennms/templates/opennms/inventory/inventory-deployment.yaml (hunk 2 above)
Comment (jaymberry) [reply charts/opennms/templates/opennms/inventory/inventory-deployment.yaml]: Strange, it is removed in the code, but the PR keeps showing it as there. Should be removed.

File: charts/opennms/templates/keycloak/keycloak-deployment.yaml [hunk 3]
```diff
@@ -61,10 +61,12 @@ spec:
             - name: KC_HOSTNAME_ADMIN_URL
               value: "{{ .Values.Keycloak.HostnameAdminUrl }}"
             {{- end }}
-            - name: KC_DB
-              value: postgres
+            - name: KC_DB_SCHEMA
+              value: "{{ .Values.Postgres.DbSchemaName }}"
```
Comment (geraldhumphries) [inline charts/opennms/templates/keycloak/keycloak-deployment.yaml:65]: Should these be under `.Values.Keycloak...`?

Comment (jaymberry) [reply charts/opennms/templates/keycloak/keycloak-deployment.yaml]: Updated

**Review by geraldhumphries — APPROVED** [review]

**Pull request merged**